drivers/storage/rackspace/storage/rackspace_storage.go

```diff
@@ -0,0 +1,946 @@
+	"github.com/emccode/libstorage/api/registry"
+	"github.com/emccode/libstorage/api/types"
+	"github.com/emccode/libstorage/drivers/storage/rackspace"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"github.com/rackspace/gophercloud/openstack/blockstorage/v1/snapshots"
+	"github.com/rackspace/gophercloud/openstack/blockstorage/v1/volumes"
+	"github.com/rackspace/gophercloud/openstack/compute/v2/extensions/volumeattach"
+	//	"github.com/rackspace/gophercloud/openstack/compute/v2/servers"
+)
+
+const (
+	providerName = "Rackspace"
+	minSize      = 75 //rackspace is 75
+)
+
+type driver struct {
+	provider           *gophercloud.ProviderClient
+	client             *gophercloud.ServiceClient
+	clientBlockStorage *gophercloud.ServiceClient
+	region             string
+	instanceID         string
+	config             gofig.Config
+	//	sync.Mutex
+}
+
+func init() {
+	registry.RegisterStorageDriver(rackspace.Name, newDriver)
+}
+
+func newDriver() types.StorageDriver {
+	return &driver{}
+}
+
+func (d *driver) Name() string {
+	return rackspace.Name
+}
+
+func (d *driver) Init(context types.Context, config gofig.Config) error {
+	//fmt.Println("rackspace_storage.Init: Initializing Rackspace driver") //DEBUG
+	d.config = config
+
+	fields := eff(map[string]interface{}{})
+	var err error
+
+	if d.instanceID, err = d.getInstanceID(); err != nil {
+		return err
+	}
+
+	fields["moduleName"] = context
+	fields["instanceId"] = d.instanceID
+
+	if d.region, err = d.getInstanceRegion(); err != nil {
+		return err
+	}
+
+	fields["region"] = d.region
+	d.region = strings.ToUpper(d.region)
+
+	authOpts := d.getAuthOptions() // <-----
+
+	fields["identityEndpoint"] = d.authURL()
+	fields["userId"] = d.userID()
+	fields["userName"] = d.userName()
+	if d.password() == "" {
+		fields["password"] = ""
+	} else {
+		fields["password"] = "******"
+	}
+	fields["tenantId"] = d.tenantID()
+	fields["tenantName"] = d.tenantName()
+	fields["domainId"] = d.domainID()
+	fields["domainName"] = d.domainName()
+
+	if d.provider, err = openstack.AuthenticatedClient(authOpts); err != nil { //<--------
+		//fmt.Println("rackspace_storage.Init: Rackspace client authentication failed") //DEBUG
+		return goof.WithFieldsE(fields,
+			"error getting authenticated client", err)
+	}
+
+	if d.client, err = openstack.NewComputeV2(d.provider,
+		gophercloud.EndpointOpts{Region: d.region}); err != nil {
+		goof.WithFieldsE(fields, "error getting newComputeV2", err)
+	}
+
+	if d.clientBlockStorage, err = openstack.NewBlockStorageV1(d.provider,
+		gophercloud.EndpointOpts{Region: d.region}); err != nil {
+		return goof.WithFieldsE(fields,
+			"error getting newBlockStorageV1", err)
+	}
+
+	log.WithFields(fields).Info("storage driver initialized")
+	//fmt.Println("rackspace_storage.Init: Rackspace storage driver initialized") //DEBUG
+	return nil
+
+	
+}
+
+// 	// Type returns the type of storage the driver provides.
```

Comment:
Hi @akiraw,

Please remove the duplication of the function signatures. If you wish to leave the comments, that is fine, but you could also install an Atom plug-in that will show you the comment as it is defined on the interface. Thanks!

p.s. This comment should be applied to wherever it is applicable in this file and others. Thanks again!